homeassistant/components/atome/sensor.py

```diff
@@ -0,0 +1,319 @@
+
+SESSION_RENEW_INTERVAL = timedelta(minutes=55)
+
+LIVE_SCAN_INTERVAL = timedelta(seconds=30)
+DAILY_SCAN_INTERVAL = timedelta(seconds=150)
+WEEKLY_SCAN_INTERVAL = timedelta(hours=1)
+MONTHLY_SCAN_INTERVAL = timedelta(hours=1)
+YEARLY_SCAN_INTERVAL = timedelta(days=1)
+
+LIVE_NAME = "Atome Live Power"
+DAILY_NAME = "Atome Daily"
+WEEKLY_NAME = "Atome Weekly"
+MONTHLY_NAME = "Atome Monthly"
+YEARLY_NAME = "Atome Yearly"
+
+LIVE_TYPE = "live"
+DAILY_TYPE = "day"
+WEEKLY_TYPE = "week"
+MONTHLY_TYPE = "month"
+YEARLY_TYPE = "year"
+
+ICON = "mdi:flash"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Atome sensor."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    data = AtomeData(username, password)
+
+    @Throttle(LIVE_SCAN_INTERVAL)
+    def update_live():
+        """Update the live power usage."""
+        data.get_live_power()
+
+    @Throttle(DAILY_SCAN_INTERVAL)
+    def update_daily():
+        """Update the daily power usage."""
+        data.get_daily_usage()
+
+    @Throttle(WEEKLY_SCAN_INTERVAL)
+    def update_weekly():
+        """Update the weekly power usage."""
+        data.get_weekly_usage()
+
+    @Throttle(MONTHLY_SCAN_INTERVAL)
+    def update_monthly():
+        """Update the monthly power usage."""
+        data.get_monthly_usage()
+
+    @Throttle(YEARLY_SCAN_INTERVAL)
+    def update_yearly():
+        """Update the yearly power usage."""
+        data.get_yearly_usage()
+
+    update_live()
+    update_daily()
+    update_weekly()
+    update_monthly()
+    update_yearly()
+
+    add_entities([AtomeSensor(data, LIVE_NAME, LIVE_TYPE, update_live)], True)
+    add_entities([AtomeSensor(data, DAILY_NAME, DAILY_TYPE, update_daily)], True)
+    add_entities([AtomeSensor(data, WEEKLY_NAME, WEEKLY_TYPE, update_weekly)], True)
+    add_entities([AtomeSensor(data, MONTHLY_NAME, MONTHLY_TYPE, update_monthly)], True)
+    add_entities([AtomeSensor(data, YEARLY_NAME, YEARLY_TYPE, update_yearly)], True)
+
+
+class AtomeData:
+    """Stores data retrieved from Neurio sensor."""
+
+    def __init__(self, username, password):
+        """Initialize the data."""
+        self.username = username
+        self.password = password
+        self._live_power = None
+        self._subscribed_power = None
+        self._is_connected = None
+        self._daily_usage = None
+        self._daily_price = None
+        self._weekly_usage = None
+        self._weekly_price = None
+        self._monthly_usage = None
+        self._monthly_price = None
+        self._yearly_usage = None
+        self._yearly_price = None
+
+        self._state = None
+
+        try:
+            self.atome_client = AtomeClient(username, password)
```

Comment:
We don't want side effects in init method. Create the client in `setup_platform`, login, then pass the client to the data instance when creating it.

Reply:
ok, changed.THX!